index.js

```diff
@@ -55,7 +54,7 @@ export default class Flatbush {
             throw new Error(`Unexpected typed array class: ${ArrayType}.`);
         }
 
-        if (data && (data instanceof ArrayBuffer)) {
+        if (data && (data.constructor.name === 'ArrayBuffer' || data.constructor.name === 'SharedArrayBuffer')) {
```

Comment:
Equivalent to the suggestion above:

```suggestion
        if (data && data.byteLength !== undefined && !data.buffer) {
```